packages/svelte/src/lib/types.ts

```diff
@@ -1,10 +1,10 @@
 import type { PageProps } from '@inertiajs/core'
 import type { ComponentType } from 'svelte'
 
-export type ComponentResolver = (name: string) => ComponentType | Promise<ComponentType>
+export type ComponentsResolver = (name: string) => ResolvedComponents | Promise<ResolvedComponents>
 
-export interface InertiaComponentType extends ComponentType {
-  default: InertiaComponentType
-  layout: InertiaComponentType
-  props: PageProps
+export type ResolvedComponents = {
+  default?: ComponentType
+  layout?: ComponentType,
+  props?: PageProps
```

Comment:
I left `props?: PageProps` but i'm honestly not sure it's actually used at all. I had it deleted in my branch.

Looking here: https://inertiajs.com/pages#default-layouts the return value looks like

```
return { default: page.default, layout: page.layout || Layout }
```

and without a layout it would be 

```
return { default: page.default }
// which is same as
return page
```

Reply:
It looks like `props` is indeed not used from what I can tell, we may as well remove it.

Reply:
Done ! 👍